kie-server-parent/kie-server-services/kie-server-services-dmn/src/main/java/org/kie/server/services/dmn/ModelEvaluatorServiceBase.java

```diff
@@ -162,11 +160,13 @@ public static DMNItemDefinitionInfo itemDefinitionToInfo(ItemDefinition itemDef)
             KieSession kieSession = kContainer.getKieContainer().newKieSession();
             DMNRuntime dmnRuntime = kieSession.getKieRuntime(DMNRuntime.class);
 
-            KieServerExtension extension = context.getServerExtension(PrometheusKieServerExtension.EXTENSION_NAME);
+            PrometheusKieServerExtension extension = (PrometheusKieServerExtension)context.getServerExtension(PrometheusKieServerExtension.EXTENSION_NAME);
             if (extension != null) {
-                PrometheusMetrics dmnMetrics = PrometheusKieServerExtension.getMetrics();
-                PrometheusMetricsDMNListener listener = new PrometheusMetricsDMNListener(dmnMetrics, kContainer);
-                dmnRuntime.addListener(listener);
+                List<DMNRuntimeEventListener> listeners = extension.getDMNRuntimeListeners(kContainer);
+                boolean listenersAlreadyAdded = dmnRuntime.getListeners().containsAll(listeners);
```

Comment:
Would it be possible for dmnRuntime to contain just some extension listeners, but not all?

Reply:
I can't see how. Either we added already all Prometheus listeners or not at all. This is more of a shortcut to re-adding listeners all the time. Only add them once bascially.

Reply:
ok